### PR TITLE
[IMS-2] Registration endpoints 

### PR DESCRIPTION
**Pull request by leahnjoroge**
Fixed the user/registration method by adding @Autowired to it

### CODE REVIEW

File: src/main/java/com/example/demo/controller/RegistrationRestController.java [hunk 1]
```diff
@@ -63,7 +65,7 @@ public GenericResponse registerUserAccount(@Valid final UserDto accountDto, fina
 
     // User activation - verification
     @GetMapping("/user/resendRegistrationToken")
-    public GenericResponse resendRegistrationToken(final HttpServletRequest request, @RequestParam("token") final String existingToken) {
```
Comment (leahnjoroge) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:66]: rename

File: src/main/java/com/example/demo/controller/RegistrationRestController.java [hunk 2]
```diff
@@ -84,7 +86,7 @@ public GenericResponse resetPassword(final HttpServletRequest request, @RequestP
 
     // Save Password
     @PostMapping("/user/savePassword")
-    public GenericResponse savePassword(final Locale locale, @Valid PasswordDto passwordDto) {
+    public GenericResponse savePassword(@RequestBody final Locale locale, @RequestBody @Valid PasswordDto passwordDto) {
```
Comment (leahnjoroge) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:90]: we can take this one out

File: src/main/java/com/example/demo/controller/RegistrationRestController.java [hunk 3]
```diff
@@ -52,7 +54,7 @@ public RegistrationRestController() {
 
     // Registration
     @PostMapping("/user/registration")
-    public GenericResponse registerUserAccount(@Valid final UserDto accountDto, final HttpServletRequest request) {
```
Comment (leahnjoroge) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:55]: rename to registration verification

File: src/main/java/com/example/demo/controller/RegistrationRestController.java [hunk 4]
```diff
@@ -103,7 +105,7 @@ public GenericResponse savePassword(final Locale locale, @Valid PasswordDto pass
 
     // Change User password
     @PostMapping("/user/updatePassword")
```
Comment (leahnjoroge) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:108]: account/recovery/password

Comment (fredthuo1) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:107]: hhghg

File: src/main/java/com/example/demo/controller/RegistrationRestController.java [hunk 5]
```diff
@@ -84,7 +86,7 @@ public GenericResponse resetPassword(final HttpServletRequest request, @RequestP
 
     // Save Password
     @PostMapping("/user/savePassword")
```
Comment (fredthuo1) [inline src/main/java/com/example/demo/controller/RegistrationRestController.java:88]: dfgdfgdfgd

File: src/main/java/com/example/demo/controller/RegistrationRestController.java (hunk 4 above)
Comment (fredthuo1) [reply src/main/java/com/example/demo/controller/RegistrationRestController.java:108]: This is right, the method changes password from old to new password.

**Pull request closed without merging**